mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/ReactorRabbitMQChannelPoolTest.java

```diff
@@ -0,0 +1,87 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.james.backend.rabbitmq.RabbitMQExtension;
+import org.apache.james.util.concurrency.ConcurrentTestRunner;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.github.fge.lambdas.Throwing;
+import com.rabbitmq.client.Channel;
+
+import reactor.core.publisher.Mono;
+import reactor.rabbitmq.ChannelPool;
+
+class ReactorRabbitMQChannelPoolTest implements ChannelPoolContract {
+
+    @RegisterExtension
+    static RabbitMQExtension rabbitMQExtension = RabbitMQExtension.singletonRabbitMQ();
+
+    private List<ReactorRabbitMQChannelPool> channelPools;
+
+    @BeforeEach
+    void beforeEach() {
+        channelPools = new ArrayList<>();
+    }
+
+    @AfterEach
+    void afterEach() {
+        channelPools.forEach(ReactorRabbitMQChannelPool::close);
+    }
+
+    @Override
+    public ChannelPool getChannelPool(int poolSize) {
+        ReactorRabbitMQChannelPool channelPool = new ReactorRabbitMQChannelPool(
+            rabbitMQExtension.getRabbitConnectionPool().getResilientConnection(),
+            poolSize);
+        channelPools.add(channelPool);
+
+        return channelPool;
+    }
+
+    // Pool wait timeout is an expected exception
+    @Test
+    void concurrentRequestOnChannelMonoLeadToPoolWaitTimeoutException() {
```

Comment:
I don't understand why this should be a desirable property

Reply:
when all objects in the pool are acquired, then if other requests to acquire channels reach to wait timeout, they will be rejected. it's pretty same the a test above but in concurrent environment

Reply:
which properties would you like with that pool?

Reply:
ok, I understand, that's ok